### ISSUE TITLE
Whitelist Subnet Broken Again in 3.18.3

### ISSUE THREAD

**Issue by kevinmarchant** [NordSecurity/nordvpn-linux#512]
After upgrading to 3.18.3 from 3.18.2 on Ubuntu 24.0.4 the problem with whitelisting subnets is back

**Comment by keliramu** [comment]
Hello, could you give us more details - what exactly is not working as expected?

**Comment by kevinmarchant** [comment]
I have a VM running Ubuntu 24.04 and Nordvpn 3.18.2.  In the NordVPN config I have

Technology: NORDLYNX
Firewall: enabled
Firewall Mark: 0xe1f1
Routing: enabled
Analytics: disabled
Kill Switch: disabled
Threat Protection Lite: disabled
Notify: disabled
Tray: enabled
Auto-connect: enabled
IPv6: disabled
Meshnet: disabled
DNS: 1.0.0.1, 1.1.1.1
LAN Discovery: disabled
Allowlisted subnets:
        192.168.1.0/24
        192.168.2.0/24

With 3.18.2 I can access access the VM from 192.168..* (VNC and http).  After upgrading top 3.18.3 it no longer works.

This is the same but in Allowlist subnets as existed in 3.18.1 and was fixed in 3.18.2.  So there has been a regression.

**Comment by keliramu** [comment]
Hello, unfortunately, we are not able to reproduce this problem you are describing. Old version 3.18.1 had this issue - I can reproduce that, but with versions 3.18.2 and 3.18.3 - I cannot reproduce this problem, my colleague as well. Even with non-VM setup.  

These are steps I'm doing after install on VM:
1. on VM: nordvpn login & nordvpn connect
2. on Host: ping VM IP - no response;
3. on VM: nordvpn allowlist add subnet <mysubnet> (that is a subnet where VM belongs);
4. on Host: ping VM IP - all ok, responding; access http server on VM as well;

@kevinmarchant could you check it once again? Also, could you provide steps how you come to the problem? Maybe I'm trying different steps?

**Comment by kevinmarchant** [comment]
ICMP (ping) does continue to work as you say, but TCP (port 80 / 8080) are blocked, i.e. accessing a web server inside the VM (in this case qBittorrent.

**Comment by keliramu** [comment]
I tried tcp port 8000 running simple http server, also on ports 8080 and 80 - it's going through

**Comment by keliramu** [comment]
Could you send us FW rules (output of `sudo iptables-save`) from VM when the problem is taking place?

**Comment by kevinmarchant** [comment]
I've just installed Apache2 in the VM and have the same results (can access from 3.18.2 with both VPN connected or disconnected) and can only access with 3.18.3 when VPN not connected.  Here is the required info.

[iptables-save.txt](https://github.com/user-attachments/files/16334750/iptables-save.txt)

**Comment by keliramu** [comment]
Thanks, FW setup looks ok. Is your host in `192.168.1.0/24` subnet? Now I want to make sure the traffic to the VM is comming from the allowed subnet.

Also, could you send me from VM an output of these: `ip rule` and `ip route show table all`?

**Comment by kevinmarchant** [comment]
I will do that - but in your testing can you confirm you have LAN Discovery = disabled, as the partial work around for the broken 3.18.1 to allow access from just one subnet was to set this setting to enabled.

Yes the host is in the 192.168.1.0/24 subnet.  The second subnet is needed to allow remote access to the web server when offsite.

**Comment by keliramu** [comment]
Yes, I can confirm LAN discovery is OFF

**Comment by kevinmarchant** [comment]
Debug info as requested.
[ip-route-show-table-all.txt](https://github.com/user-attachments/files/16335459/ip-route-show-table-all.txt)
[ip-rule.txt](https://github.com/user-attachments/files/16335461/ip-rule.txt)

**Comment by keliramu** [comment]
Thanks, also all looks OK...

**Comment by donbernhardo** [comment]
I have the same issue with 3.18.3.
I set the local subnet 192.168.2.0/24 to be whitelisted and as soon as i connect to nrodvpn the client looses all connection to the local network. for example the ssh connection breaks and the samba mounted directories become unreachable.

reverting back to 3.18.2 and everything works again.

**Comment by sprokkel78** [comment]
Same here, whitelisting subnet 192.168.1.0/24 doesn't work because I can't ping the LAN anymore. If I put LAN-discovery ON then I can ping the LAN again.

**Comment by keliramu** [comment]
Thank you for a patience, @kevinmarchant, plz could you help us debug this problem on your setup (as we are not able to reproduce), with new v3.18.3 installed and vpn connected and with allowlist subnet added 192.168.1.0/24, could you modify routes:
```
sudo ip route del 192.168.1.0/24 via 192.168.1.1 dev ens18 table 205
sudo ip route add 192.168.1.0/24 dev ens18 table 205 scope link
```
And tell if that does change anything.

**Comment by kevinmarchant** [comment]
Yes, that seem to resolve the issue.  I can SSH into the VM and access web services as per 3.18.2.   I hope this will allow you to debug at your end ?

**Comment by keliramu** [comment]
Thank you, your info helps a lot.

**Comment by bsteiner36** [comment]
I just upgraded to 3.18.3 and nordvpn does connect but my whitelisting does not seem to work for incoming connections other than pings.  I can ping in and out but I can not ssh to the server running nordvpn.  I have the subnets in the whitelist/allowlist.

**Comment by keliramu** [comment]
Sorry for an inconvenience, now we have identified the root cause and working on fix and planning to include this fix in upcoming release.

**Comment by kevinmarchant** [comment]
That’s good thanks.  If was a regression rather than a new bug, perhaps you could enhance your test suite to ensure it doesn’t reoccur?

**Comment by keliramu** [comment]
@kevinmarchant, issue fixed - please check new release 3.18.4, will be waiting for your feedback.

**Comment by kevinmarchant** [comment]
Issue resolved!  Thanks.
